### PR TITLE
Inform user if new space looks like old space

### PR DESCRIPTION
**Pull request by DanielJackson-Oslo**
## Checks for duplicates and shows them to the user like this when creating a new space

image

Checks based on address, post_number and title in a service.

## Not solved in this branch, should be added as cards on Trello:

- [x] Stopping submits before the check has been ran, and then  re-enable when results are in. Not sure how to do this! Seems like Rails stops form submits after the first atttempt, even if the first form submit was stopped with preventDefault.
- [ ] Handle if the service does not respond. Right now the form just won't be submittable if the duplicate checker service crashes or does not respond.
- [ ] Error handling for Space#new (handled in @mathias234 capybara branch)

(Based on #135 to get styling right for when I make the form again)

### CODE REVIEW

File: app/services/spaces/duplicate_detector_service.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module Spaces
+  class DuplicateDetectorService < ApplicationService
+    def initialize(space)
+      @title = space.title
+      @address = space.address
+      @post_number = space.post_number
+      super()
+    end
```
Comment (mathias234) [inline app/services/spaces/duplicate_detector_service.rb:10]: This service looks pretty good although convention in the ruby world is to add attr_readers to each of the fields so 
`attr_reader :title, :address, :space` a proper guide for this here https://www.rubyguides.com/2018/11/attr_accessor/

File: app/services/spaces/duplicate_detector_service.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      # Then return any matches by title and/or post
+      duplicates << by_post_and_title
+      duplicates.flatten.compact.uniq
+    end
+
+    def full_match
+      return nil unless @title.present? && @address.present? && @post_number.present?
+
+      Space.where(
+        "title LIKE ?", "%#{@title}%"
```
Comment (mathias234) [inline app/services/spaces/duplicate_detector_service.rb:39]: Could do `"title ILIKE ?", "%#{@title}%"` instead to match without case sensitivity

**Review by mathias234 — CHANGES_REQUESTED** [review]
Looks very good, perfect use case for a service! Had a few notes on the the service itself (see individual comments) but nothing major. 

Other things: 

 * _Stopping submits before the check has been ran, and then re-enable when results are in. Not sure how to do this! Seems like Rails stops form submits after the first atttempt, even if the first form submit was stopped with preventDefault._
 This is because rails always enables `data-disable-with` on submit buttons you can add `data: { disable_with: false },` to the submit button to disable this feature

File: app/services/spaces/duplicate_detector_service.rb (hunk 1 above)
Comment (DanielJackson-Oslo) [reply app/services/spaces/duplicate_detector_service.rb:10]: Would that also be needed when I don't want to expose them to the outside world? These aren't used by any external resources calling the service.

File: app/services/spaces/duplicate_detector_service.rb (hunk 2 above)
Comment (DanielJackson-Oslo) [reply app/services/spaces/duplicate_detector_service.rb:39]: Sweet! Will fix

File: app/services/spaces/duplicate_detector_service.rb (hunk 1 above)
Comment (DanielJackson-Oslo) [reply app/services/spaces/duplicate_detector_service.rb:10]: Fixed!

**Review by mathias234 — APPROVED** [review]
LGTM!

**Pull request merged**